rest_api/api/offers.py

```diff
@@ -53,3 +85,21 @@
 async def close_offer(request, offer_id):
     """Request by owner of Offer to close it"""
     raise ApiNotImplemented()
+
+
+def _create_offer_dict(body, public_key):
+    keys = ['label', 'description', 'source', 'rules',
+            'sourceQuantity', 'target', 'targetQuantity']
+
+    offer = {k: body[k] for k in keys if body.get(k) is not None}
+
+    if offer.get('sourceQuantity') is None:
+        offer['sourceQuantity'] = 0
+    if offer.get('targetQuantity') is None:
+        offer['targetQuantity'] = 0
```

Comment:
I'm not sure this correct given our recent Slack discussion. A source quantity of 0 is never valid, and a target quantity of 0 is only valid if target is also unspecified. I think it is probably best to leave these fields unset. In most cases, the validator will spit back an invalid transaction anyway. In the event that both target and targetQuantity are unset, leaving them unset in the response object seems sensible to me.